From a48c93ace59c572f06c853633e479946834edda4 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:18:25 -0400
Subject: [fs] nfsd: Avoid warning message when compile at i686 arch

Message-id: <1410898849-22126-22-git-send-email-bfields@redhat.com>
Patchwork-id: 95701
O-Subject: [Patch RHEL7 021/165] NFSD: Avoid warning message when compile at i686 arch
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Kinglong Mee <kinglongmee@gmail.com>

fs/nfsd/nfs4xdr.c: In function 'nfsd4_encode_readv':
>> fs/nfsd/nfs4xdr.c:3137:148: warning: comparison of distinct pointer types lacks a cast [enabled by default]
thislen = min(len, ((void *)xdr->end - (void *)xdr->p));

Reported-by: Fengguang Wu <fengguang.wu@intel.com>
Signed-off-by: Kinglong Mee <kinglongmee@gmail.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 1055414fe19db2db6c8947c0b9ee9c8fe07beea1
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4xdr.c b/fs/nfsd/nfs4xdr.c
index 2f248e2..decf56f 100644
--- a/fs/nfsd/nfs4xdr.c
+++ b/fs/nfsd/nfs4xdr.c
@@ -3140,7 +3140,7 @@ static __be32 nfsd4_encode_readv(struct nfsd4_compoundres *resp,
  len = maxcount;
  v = 0;
 
- thislen = min(len, ((void *)xdr->end - (void *)xdr->p));
+ thislen = min_t(long, len, ((void *)xdr->end - (void *)xdr->p));
  p = xdr_reserve_space(xdr, (thislen+3)&~3);
  WARN_ON_ONCE(!p);
  resp->rqstp->rq_vec[v].iov_base = p;
-- 
1.7.1